api/publish.go

```diff
@@ -179,7 +181,15 @@ func apiPublishRepoOrSnapshot(c *gin.Context) {
 
 	taskName := fmt.Sprintf("Publish %s: %s", b.SourceKind, strings.Join(names, ", "))
 	task, conflictErr := runTaskInBackground(taskName, resources, func(out *task.Output, detail *task.Detail) error {
-		published.Origin = b.Origin
+		if b.Origin != "" {
```

Comment:
Why was this logic changed from conditionless assignment to only assigning when nonempty?